File: repos_estudiantes_ALVAREZ_segunda_entrega/AGUILAR_MACHICADO_NILS_REINER/PRACTICA 2/HERENCIA/Ejercicio1.py
class Vehiculo:
    def __init__(self, marca, modelo, anio, precio_base):
        self.marca = marca
        self.modelo = modelo
        self.anio = anio
        self.precio_base = precio_base

    def mostrar_info(self):
        return f"Marca: {self.marca}, Modelo: {self.modelo}, Año: {self.anio}, Precio Base: {self.precio_base}"

class Coche(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, num_puertas, tipo_combustible):
        super().__init__(marca, modelo, anio, precio_base)
        self.num_puertas = num_puertas
        self.tipo_combustible = tipo_combustible

    def mostrar_info(self):
        return super().mostrar_info() + f", Puertas: {self.num_puertas}, Combustible: {self.tipo_combustible}"

class Moto(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, cilindrada, tipo_motor):
        super().__init__(marca, modelo, anio, precio_base)
        self.cilindrada = cilindrada
        self.tipo_motor = tipo_motor

    def mostrar_info(self):
        return super().mostrar_info() + f", Cilindrada: {self.cilindrada}, Tipo Motor: {self.tipo_motor}"

vehiculos = [
    Coche("Toyota", "Corolla", 2025, 15000, 4, "Gasolina"),
    Coche("Ford", "Explorer", 2024, 20000, 5, "Diésel"),
    Moto("Honda", "CB500", 2025, 8000, 500, "4 tiempos"),
    Moto("Yamaha", "R15", 2023, 6000, 155, "2 tiempos")
]

print("--- Todos los Vehículos ---")
for v in vehiculos:
    print(v.mostrar_info())

print("\n--- Coches con más de 4 puertas ---")
for v in vehiculos:
    if isinstance(v, Coche) and v.num_puertas > 4:
        print(v.mostrar_info())

print("\n--- Vehículos del año 2025 ---")
for v in vehiculos:
    if v.anio == 2025:
        print(v.mostrar_info())

File: repos_estudiantes_ALVAREZ_segunda_entrega/AGUILAR_MACHICADO_NILS_REINER/PRACTICA 2/HERENCIA/Ejercicio1.java
class Vehiculo {
    private String marca;
    private String modelo;
    private int anio;
    private double precioBase;

    public Vehiculo(String marca, String modelo, int anio, double precioBase) {
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
        this.precioBase = precioBase;
    }

    public int getAnio() {
        return anio;
    }

    public String toString() {
        return "Marca: " + marca + ", Modelo: " + modelo + ", Año: " + anio + ", Precio Base: " + precioBase;
    }
}

class Coche extends Vehiculo {
    private int numPuertas;
    private String tipoCombustible;

    public Coche(String marca, String modelo, int anio, double precioBase, int numPuertas, String tipoCombustible) {
        super(marca, modelo, anio, precioBase);
        this.numPuertas = numPuertas;
        this.tipoCombustible = tipoCombustible;
    }

    public int getNumPuertas() {
        return numPuertas;
    }

    public String toString() {
        return super.toString() + ", Puertas: " + numPuertas + ", Combustible: " + tipoCombustible;
    }
}

class Moto extends Vehiculo {
    private int cilindrada;
    private String tipoMotor;

    public Moto(String marca, String modelo, int anio, double precioBase, int cilindrada, String tipoMotor) {
        super(marca, modelo, anio, precioBase);
        this.cilindrada = cilindrada;
        this.tipoMotor = tipoMotor;
    }

    public String toString() {
        return super.toString() + ", Cilindrada: " + cilindrada + ", Tipo Motor: " + tipoMotor;
    }
}

class TestVehiculos {
    public static void main(String[] args) {
        Vehiculo[] vehiculos = new Vehiculo[] {
            new Coche("Toyota", "Corolla", 2025, 15000, 4, "Gasolina"),
            new Coche("Ford", "Explorer", 2024, 20000, 5, "Diésel"),
            new Moto("Honda", "CB500", 2025, 8000, 500, "4 tiempos"),
            new Moto("Yamaha", "R15", 2023, 6000, 155, "2 tiempos")
        };

        System.out.println("--- Todos los Vehículos ---");
        for (Vehiculo v : vehiculos) {
            System.out.println(v);
        }

        System.out.println("\n--- Coches con más de 4 puertas ---");
        for (Vehiculo v : vehiculos) {
            if (v instanceof Coche && ((Coche)v).getNumPuertas() > 4) {
                System.out.println(v);
            }
        }

        System.out.println("\n--- Vehículos del año 2025 ---");
        for (Vehiculo v : vehiculos) {
            if (v.getAnio() == 2025) {
                System.out.println(v);
            }
        }
    }
}